lib/active_record/connection_adapters/sqlserver_adapter.rb

```diff
@@ -333,6 +333,8 @@ def translate_exception(e, message)
           NoDatabaseError.new(message)
         when /data would be truncated/
           ValueTooLong.new(message)
+        when /connection timed out/
```

Comment:
Better match the full message `Adaptive Server connection timed out`. What do you think?

Reply:
The question is whether we use `StatementInvalid` or `StatementTimeout` when we encounter `connection timed out` String.

In my opinion, when `/time(?:d )?out/i` shows up it is very likely that `StatementTimeout` would be more accurate than `StatementInvalid`.

It depends on how we prioritize if we prioritize translation, loosely matching the String should be better. On the other hand, if we prioritize known cases and knowledge, matching the String exactly would be a better choice.

As a maintainer, which one you prioritize?

Reply:
Ok, makes sense. Let's leave as it is.